website_sale_slides_order_line_link/models/slide_channel.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2025 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SlideChannelPartner(models.Model):
+    _inherit = "slide.channel.partner"
+
+    sale_order_line_ids = fields.One2many(
+        comodel_name="sale.order.line",
+        inverse_name="channel_partner_id",
+        string="Sale Order Lines",
+    )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        so_line_ids = self.env.context.get("sale_order_lines")
+        if so_line_ids:
+            sale_order_lines = self.env["sale.order.line"].browse(so_line_ids)
+            channel_ids = {
+                vals["channel_id"] for vals in vals_list if "channel_id" in vals
+            }
+            channels = self.env["slide.channel"].browse(channel_ids)
+            channel_product_map = {c.id: c.product_id.id for c in channels}
+            product_line_map = {
+                line.product_id.id: line.id for line in sale_order_lines
+            }
+            for vals in vals_list:
+                product_id = channel_product_map.get(vals.get("channel_id"))
+                matching_line_id = product_line_map.get(product_id)
+                if matching_line_id:
+                    vals["sale_order_line_ids"] = [(6, 0, [matching_line_id])]
+        return super().create(vals_list)
```

Comment:
It would be easier to work with the recordset after the creation of the slide channel, don't you think? Also, I think we should match the line partner with channel one.

Reply:
Done